android/jni/com/mapswithme/maps/Framework.cpp

```diff
@@ -891,6 +894,47 @@ Java_com_mapswithme_maps_Framework_nativeGetRouteFollowingInfo(JNIEnv * env, jcl
   return result;
 }
 
+JNIEXPORT jintArray JNICALL
+Java_com_mapswithme_maps_Framework_nativeGenerateRouteAltitudeChartBits(JNIEnv * env, jclass, jint width, jint height)
+{
+  ::Framework * fr = frm();
+  ASSERT(fr, ());
+  vector<uint8_t> imageRGBAData;
+
+  if (!fr->GenerateRouteAltitudeChart(width, height, imageRGBAData))
+  {
+    LOG(LWARNING, ("Can't generate route altitude image."));
+    return nullptr;
+  }
+
+  size_t const imageRGBADataSize = imageRGBAData.size();
+  ASSERT_NOT_EQUAL(imageRGBADataSize, 0, ("GenerateRouteAltitudeChart returns true but the vector with altitude image bits is empty."));
+
+  size_t const pxlCount = width * height;
+  if (maps::kAltitudeChartBPP * pxlCount != imageRGBADataSize)
+  {
+    LOG(LWARNING, ("Wrong size of vector with altitude image bits. Expected size:", pxlCount, ". Real size:", imageRGBADataSize));
+    return nullptr;
+  }
+
+  jintArray imageRGBADataArray = env->NewIntArray(imageRGBADataSize);
+  ASSERT(imageRGBADataArray, ());
+  jint * arrayElements = env->GetIntArrayElements(imageRGBADataArray, 0);
+  ASSERT(arrayElements, ());
+
+  for (size_t i = 0; i < imageRGBADataSize; ++i)
+  {
+    size_t const shiftInBytes = i * maps::kAltitudeChartBPP;
```

Comment:
optional
for (size_t i = 0, shiftInBytes = 0; i < imageRGBADataSize; ++i, shiftInBytes += maps::kAltitudeChartBPP)
to replace mult by add.